### PR TITLE
DOC: make some github issues into links, doc fixes

### LINKED ISSUE #3583: Document new features in Cython 3.0
It's time to look through the documentation to see if all new features have been documented and/or are findable for users in an appropriate place. This is a great beginners ticket because it allows playing with new Cython features and figuring out how they work.

- look through [CHANGES.rst](https://github.com/cython/cython/blob/master/CHANGES.rst) to see what features are new in the latest 3.0 alpha versions and which seem to be worth mentioning
- look in the documentation where you would expect this to be documented
- if it's already covered, perfect
- otherwise, search the documentation to see if it's mentioned elsewhere, and decide if that's the right place
- if not, write a paragraph or two and/or add a cross-ref from places where you think it's also worth mentioning
- if you add example code (examples are great!) make sure it compiles by adding it as a `.pyx` file under `docs/examples/` to let the test suite pick it up
- preferably, use separate commits/updates for the different features.

### PR DESCRIPTION
**Pull request by mattip**
Closes #3583 

The major things mentioned in that issue have been documented. I went through the 3.0 release notes and added a few more doc changes that seemed appropriate. As long as I was there I turned the PEP and github issue mentions into links. I can revert this if it seems too much.

There are many warnings when building docs.  I quieted some of the easiest ones:
- added the index for docs/reference and the contributing page to the front index to make them more prominent and quiet some warnings about "page not included in TOC"
- added `:orphan:` to some pages that are used as historical reference to quiet warnings about "page not included in TOC".

Fixing others warnings is more invasive so will need another PR

### CODE REVIEW

File: docs/src/donating.rst [hunk 1]
```diff
@@ -1,3 +1,5 @@
+:orphan:
+
```
Comment (mattip) [inline docs/src/donating.rst:2]: This page is linked via the top navbar so it does not need to be included in a TOC

File: Cython/Build/Dependencies.py [hunk 2]
```diff
@@ -948,6 +948,11 @@ def cythonize(module_list, exclude=None, nthreads=0, aliases=None, quiet=False,
                      See examples in :ref:`determining_where_to_add_types` or
                      :ref:`primes`.
 
+
+    :param annotate-fullc: If ``True`` will produce a colorized HTML version of
+                           the source which includes entire generated C/C++-code.
+
+
```
Comment (mattip) [inline Cython/Build/Dependencies.py:955]: This was the only real fix for new features: adding the new parameter to the "help"

File: CHANGES.rst [hunk 3]
```diff
@@ -133,43 +146,56 @@ Other changes
...
+.. _3625: https://github.com/cython/cython/issues/3625
+.. _3664: https://github.com/cython/cython/issues/3664
+.. _3616: https://github.com/cython/cython/issues/3616
+.. _3693: https://github.com/cython/cython/issues/3693
+.. _3707: https://github.com/cython/cython/issues/3707
+.. _3514: https://github.com/cython/cython/issues/3514
+.. _3663: https://github.com/cython/cython/issues/3663
+.. _3754: https://github.com/cython/cython/issues/3754
+.. _3637: https://github.com/cython/cython/issues/3637
+.. _3736: https://github.com/cython/cython/issues/3736
```
Comment (scoder) [inline CHANGES.rst:161]: Why aren't these at the end of the file as well?

On a related note – I'm not sure if I really want to maintain this list whenever I add an entry. Including merge conflicts and all that. The linking seems like something that a Sphinx plugin should do for me.

File: CHANGES.rst [hunk 4]
```diff
@@ -498,39 +524,139 @@ Other changes
...
+.. _`PEP-570`: https://www.python.org/dev/peps/pep-0570
+.. _`PEP-487`: https://www.python.org/dev/peps/pep-0487
+.. _`PEP-590`: https://www.python.org/dev/peps/pep-0590
+.. _`PEP-3131`: https://www.python.org/dev/peps/pep-3131
+.. _`PEP-563`: https://www.python.org/dev/peps/pep-0563
+.. _`PEP-479`: https://www.python.org/dev/peps/pep-0479
+
+.. _0.29.22:
 
 0.29.22 (2021-02-20)
```
Comment (scoder) [inline CHANGES.rst:661]: If any, then all versions should have an anchor like this. Again, might be something that Sphinx could do.

**Review by scoder — COMMENTED** [review]
Many nice changes, thanks. I'm not so happy with the manual changelog linking, though.

File: CHANGES.rst (hunk 4 above)
Comment (mattip) [reply CHANGES.rst:661]: The link without the anchor is `0.29.22-(2021-02-20)`, I could change the reference to that text instead.

File: CHANGES.rst (hunk 3 above)
Comment (mattip) [reply CHANGES.rst:161]: > The linking seems like something that a Sphinx plugin should do for me.

There is [sphix-issues](https://github.com/sloria/sphinx-issues), I will add it to the doc build. Then using `:issue:`3736` should automatically link to the issue.

Comment (mattip) [reply CHANGES.rst:161]: hmm, where *is* the doc build? I can't find a CI run that builds the docs. Does the readthe docs "advanced configuration" specify a `requirements.txt` ? (It should be at https://readthedocs.org/dashboard/cython/advanced/)?

File: CHANGES.rst (hunk 4 above)
Comment (mattip) [reply CHANGES.rst]: I dropped the anchors and the references. It is not worth the effort to mess with this: sphinx does not do a good job creating anchors with only digits, periods, spaces, and parentheses.

File: CHANGES.rst (hunk 3 above)
Comment (scoder) [reply CHANGES.rst:148]: These are leftovers then, right?

> where is the doc build

It's probably set up somewhat manually on readthedocs. @robertwb originally created it.

**Comment by scoder** [comment]
Looks good to me overall. I'd like to get an opinion from @robertwb of how to integrate this into the current readthedocs build.

File: CHANGES.rst (hunk 3 above)
Comment (mattip) [reply CHANGES.rst]: > These are leftovers then, right?

Removing

**Comment by mattip** [comment]
Whoever can see https://readthedocs.org/dashboard/cython/advanced/ should have permissions to add the requirements.txt.

**Comment by scoder** [comment]
I'll merge this in order to prevent it from getting stale.
Thanks!

**Pull request merged**